app/views/contexts/calculated_summary_context.py

```diff
@@ -298,20 +294,34 @@ def _get_answer_format(self, groups: Iterable[Mapping]) -> Tuple[dict, list]:
                             "unit": answer.get("unit"),
                             "unit_length": answer.get("unit_length"),
                             "currency": answer.get("currency"),
+                            "decimal_places": 0,
                         }
+
+                    self._set_decimal_places(answer, answer_format)
+
                     answer_value = answer.get("value") or 0
                     values_to_calculate.append(answer_value)
 
         return answer_format, values_to_calculate
 
+    @staticmethod
+    def _set_decimal_places(answer: Mapping, answer_format: dict) -> None:
+        if decimal_places := answer.get("decimal_places"):
+            if decimal_places > answer_format["decimal_places"]:
+                answer_format["decimal_places"] = decimal_places
```

Comment:
Once the above is done, this will need updating to handle decimal place 0 as it will be falsely. I think it would be better to do it with a single var in the main method but will leave it up to you as long as the None is default.

```py
    def _get_answer_format(self, groups: Iterable[Mapping]) -> Tuple[dict, list]:
        ...
        decimal_limits = []

        for group in groups:
            for block in group["blocks"]:
                ...
                for answer in question["answers"]:
                    ...

                    if (decimal_places := answer.get("decimal_places")) is not None:
                        decimal_limits.append(decimal_places)

        answer_format["decimal_places"] = max(decimal_limits, default=None)
        return answer_format, values_to_calculate
```

Reply:
Agree - prefer single var in method too